src/snmalloc/backend_helpers/pagemap.h

```diff
@@ -117,31 +117,35 @@ namespace snmalloc
 #endif
       SNMALLOC_ASSERT(s != 0);
       // TODO take account of pagemap size in the calculation of how big it
-      // needs to be.
+      // needs to be.  The following code creates a pagemap that covers the
+      // pagemap as well as the left over. This is not ideal, and we should
+      // really calculate the division with
+      //
+      //  bits::one_at_bit(GRANULARITY_BITS) + sizeof(T)
+      //
+      // There are awkward corner cases for the alignment of the start and
+      // the end that are hard to calculate. So this is not currently done.
 
-      // Align the start and end.  We won't store for the very ends as they
-      // are not aligned to a chunk boundary.
-      auto heap_base = pointer_align_up(b, bits::one_at_bit(GRANULARITY_BITS));
-      auto end = pointer_align_down(
+      // Calculate range in pagemap that is associated to this space.
+      // Over calculate to cover any unaligned parts at either end.
+      auto b_align = pointer_align_down(b, bits::one_at_bit(GRANULARITY_BITS));
```

Comment:
While this might take `b_align` below `b`, and so logically start our `PageMap` out of bounds, that's OK because we never access the `MetaEntry`-s associated with the `PageMap` itself?

Reply:
Is b_align only used in address and size computations? If yes, then would it be clearer and safer to get the address of b and align the address down? That way no wild pointer exists.

Reply:
So I have tried to make this capability less code as it is just addresses like you suggest:
https://github.com/microsoft/snmalloc/pull/558/commits/9542fbe931e99ec934978529142788dd0cd61c02